src/sagemaker/workflow/steps.py

```diff
@@ -524,6 +570,10 @@ def __init__(
         super(ProcessingStep, self).__init__(
             name, StepTypeEnum.PROCESSING, display_name, description, depends_on, retry_policies
         )
+        if not (run_args is None or processor is None):
+            raise ValueError("either run_args or processor need to be given.")
+
```

Comment:
The case is missing that both `run_args` and `processor` are None.
May need to check other steps as well.

Reply:
yep, fixed.